client/main.lua

```diff
@@ -713,6 +739,40 @@ function CarjackVehicle(target)
     end)
 end
 
+-- Continuous check for aiming at vehicles
+CreateThread(function()
+    while true do
```

Comment:
What is the performance of this thread?

Reply:
This thread runs every 100ms but looking back could be further optimized.

CreateThread(function()
    while true do
        local sleep = 1000 -- Default to 1 second checks when not relevant
        
        if Config.CarJackEnable and canCarjack then
            local ped = PlayerPedId()
            local hasWeapon = IsPedArmed(ped, 4) -- Check if armed with any gun
            
            if hasWeapon then
                sleep = 100 -- Check more frequently when armed
                local playerid = PlayerId()
                local aiming, target = GetEntityPlayerIsFreeAimingAt(playerid)
                
                if aiming and target ~= nil and target ~= 0 then
                    -- Rest of the carjacking logic...
                end
            end
        end
        
        Wait(sleep)
    end
end)

Only runs frequently (100ms) when the player has a weapon out
Runs much less frequently (1000ms) when unarmed
Uses IsPedArmed check which is more efficient than checking aiming every tick
Still maintains responsive carjacking when conditions are met